devito/passes/clusters/misc.py

```diff
@@ -72,30 +73,135 @@ def callback(self, clusters, prefix):
         return lifted + processed
 
 
-@timed_pass()
-def fuse(clusters):
+class Fusion(Queue):
+
     """
-    Fuse sub-sequences of Clusters with compatible IterationSpace.
+    Fuse Clusters with compatible IterationSpace.
     """
-    key = lambda c: (set(c.itintervals), c.guards)
 
-    processed = []
-    for k, g in groupby(clusters, key=key):
-        maybe_fusible = list(g)
+    def __init__(self, toposort):
+        super(Fusion, self).__init__()
+        self.toposort = toposort
```

Comment:
private `_`?

Reply:
Hum method already call like that ok

Reply:
yeah can make private but this is an inside class in the guts of the compiler, so possible sure but IMHO not really necessary